R/CohortRelationship.R

```diff
@@ -327,25 +201,60 @@ executeCohortRelationshipDiagnostics <- function(connection,
                                                  temporalCovariateSettings,
                                                  minCellCount,
                                                  recordKeepingFile,
-                                                 incremental) {
+                                                 incremental,
+                                                 batchSize = 500) {
   ParallelLogger::logInfo("Computing Cohort Relationship")
   startCohortRelationship <- Sys.time()
-
-  subset <- subsetToRequiredCohorts(
-    cohorts = cohortDefinitionSet,
+  
+  allCohortIds <- cohortDefinitionSet %>%
+    dplyr::select(.data$cohortId, .data$checksum) |>
+    dplyr::rename(targetCohortId = .data$cohortId,
+                  targetChecksum = .data$checksum) |>
+    dplyr::distinct()
+  combinationsOfPossibleCohortRelationships <- allCohortIds |>
+    tidyr::crossing(allCohortIds |>
+                      dplyr::rename(comparatorCohortId = .data$targetCohortId,
+                                    comparatorChecksum = .data$targetChecksum)) |>
+    dplyr::filter(.data$targetCohortId != .data$comparatorCohortId) |> 
+    dplyr::arrange(.data$targetCohortId, .data$comparatorCohortId) |> 
+    dplyr::mutate(checksum = paste0(.data$targetChecksum, .data$comparatorChecksum))
+  
+  subset <- subsetToRequiredCombis(
+    combis = combinationsOfPossibleCohortRelationships,
     task = "runCohortRelationship",
     incremental = incremental,
     recordKeepingFile = recordKeepingFile
   )
 
   if (nrow(subset) > 0) {
     if (incremental &&
-      (nrow(cohortDefinitionSet) - nrow(subset)) > 0) {
-      ParallelLogger::logInfo(sprintf(
-        " - Skipping %s cohort combinations in incremental mode.",
-        nrow(cohortDefinitionSet) - nrow(subset)
-      ))
+        (nrow(cohortDefinitionSet) - (length(subset$targetCohortId |> unique()))) > 0) {
+      ParallelLogger::logInfo(
+        sprintf(
+          " - Skipping %s target cohorts in incremental mode because the relationships has already been computed with other cohorts.",
+          nrow(cohortDefinitionSet) - (length(subset$targetCohortId |> unique()))
```

Comment:
Again, avoid use of native pipe for consistency across the package. The behaviour is different to mittagr and this can lead to confusion/errors